asf_levies_model/consumers.py

```diff
@@ -0,0 +1,460 @@
+            The fuel poverty gap estimate, in £.
+        """
+        total_fuel_spend = (
+            self.electricity_bill + self.gas_bill + self.unmetered_fuel_spend
+        )
+        threshold = 0.1 * self.net_annual_income
+        return max(total_fuel_spend - threshold, 0)
+
+    @property
+    def electricity_subtotal_bill(self) -> float:
+        """Get the electricity subtotal bill (before any social support adjustments).
+
+        Returns
+        -------
+        float
+            Annual electricity subtotal bill, including VAT, in £.
+        """
+        return self.electricity_tariff.calculate_total_consumption(
+            self.electricity_consumption, vat=True
+        )
+
+    @property
+    def gas_subtotal_bill(self) -> float:
+        """Get the gas subtotal bill (before any social support adjustments).
+
+        Returns
+        -------
+        float
+            Annual gas subtotal bill, including VAT, in £.
+        """
+        return self.gas_tariff.calculate_total_consumption(
+            self.gas_consumption, vat=True
+        )
+
+    @property
+    def electricity_bill(self) -> float:
+        """Get the final electricity bill (after social support adjustments, if any). If no social support is implemented, the final bill is the same as the subtotal bill.
+
+        Returns
+        -------
+        float
+            Annual electricity bill, including VAT, in £.
+        """
+        if self._electricity_bill is not None:
+            return self._electricity_bill
+        return self.electricity_subtotal_bill
+
+    @electricity_bill.setter
+    def electricity_bill(self, value: float) -> None:
+        """Set the final electricity bill value.
+
+        Parameters
+        ----------
+        value : float
+            The new final electricity bill value, in £.
+        """
+        self._electricity_bill = value
+
+    @property
+    def gas_bill(self) -> float:
+        """Get the final gas bill (after social support adjustments, if any). If no social support is implemented, the final bill is the same as the subtotal bill.
+
+        Returns
+        -------
+        float
+            Annual gas bill, including VAT, in £.
+        """
+        if self._gas_bill is not None:
+            return self._gas_bill
+        return self.gas_subtotal_bill
+
+    @gas_bill.setter
+    def gas_bill(self, value: float) -> None:
+        """Set the final gas bill value.
+
+        Parameters
+        ----------
+        value : float
+            The new final gas bill value, in £.
+        """
+        self._gas_bill = value
+
+    @property
+    def dual_fuel_bill(self) -> float:
+        """Get the combined final bill for electricity and gas (after any social support adjustment).
+
+        Returns
+        -------
+        float
+            Annual dual fual bill, including VAT, in £.
+        """
+        return self.gas_bill + self.electricity_bill
+
+    def apply_social_support_adjustment(
+        self,
```

Comment:
This is fine for now, but we can refactor later to make it more DRY. Currently gas and electricity are conditionals, but you could abstract this away (e.g. with a dictionary or function).

Reply:
We might also look to make it return a copy of the object (by default) and only make the change in place by argument (`inplace`). Currently this only does an inplace update.